midx.c

```diff
@@ -449,6 +449,8 @@ struct pack_list {
 	uint32_t nr;
 	uint32_t alloc;
 	struct multi_pack_index *m;
+	struct progress *progress;
+	uint32_t pack_paths_checked;
```

Comment:
@derrickstolee this was the change I wasn't 100% sure about.  

`pack_list` is only used by `add_pack_to_midx` and adding `progress` and `pack_paths_checked` to the struct was the best way I could find to track the state of our progress across calls.  If you can think of any alternative you'd prefer (or if you think `pack_list` should be renamed to something more specific to `add_pack_to_midx`) please let me know.

(I also considered not even calling `display_progress` in this path and just having start/stop, but that seemed like a worse option.  There's one spot we're doing that with a `QSORT` and there's a `TODO` there to measure its progress).

Reply:
This `pack_list` struct is an internal struct just for passing to these iteration methods. This is absolutely the right place to put the progress pointer and progress count.

QSORT is hard to add progress in, because its internal code isn't progress-aware. It is possible to create a progress indicator, but it will never update, just print "Sorting objects...done".

What seems to be missing, but _may_ be interesting, is a "Writing objects to multi-pack-index" progress, particularly during the chunk that writes the objects. Or, just have a non-numbered progress for "Writing multi-pack-index...".